doc/src/manual/performance-tips.md

```diff
@@ -129,23 +135,36 @@ abstract types where possible.
 
 Consider the following:
 
-```julia
-a = Real[]    # typeof(a) = Array{Real,1}
-if (f = rand()) < .8
```

Comment:
Changed this example to show that a `Real[]` can store a bunch of different random stuff to better make the point that it must store things "inefficently". Contrast that with a `Float64[]` below.